homework_4/bets_api/internal/domain/services/betService.go

```diff
@@ -0,0 +1,41 @@
+package services
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/jelenanemcic/code-cadets-2021/homework_4/bets_api/internal/domain/models"
+	"github.com/jelenanemcic/code-cadets-2021/homework_4/bets_api/internal/infrastructure/sqlite"
+	"github.com/pkg/errors"
+)
+
+// BetService implements bet related functions.
+type BetService struct {
+	betRepository sqlite.BetRepository
+}
+
+// NewBetService creates a new instance of BetService.
+func NewBetService(repository sqlite.BetRepository) *BetService {
+	return &BetService{
+		betRepository: repository,
+	}
+}
+
+//GetBetById returns a bet with the specific id.
+func (e BetService) GetBetById(ctx *gin.Context) (models.BetReduced, bool, error) {
+	id := ctx.Param("id")
```

Comment:
`BetService` should contain business logic only but by accepting `*gin.Context` it got coupled with `gin` and it "knows" it's being used as part of the HTTP API.

I would suggest moving `id := ctx.Param("id")` into the controller logic and changing this method to:
`func (e BetService) GetBetById(ctx context.Context, id string) (models.BetReduced, bool, error)`

Similar for other methods as well.

Reply:
I fixed it :)